SU2_CFD/src/drivers/CDriver.cpp

```diff
@@ -1027,32 +1029,38 @@ void CDriver::Solver_Preprocessing(CConfig* config, CGeometry** geometry, CSolve
   fem_dg_flow      = false;  fem_dg_shock_persson = false;
   e_spalart_allmaras = false; comp_spalart_allmaras = false; e_comp_spalart_allmaras = false;
   
-  bool compressible   = (config->GetKind_Regime() == COMPRESSIBLE);
-  bool incompressible = (config->GetKind_Regime() == INCOMPRESSIBLE);
+  bool compressible   = false;
```

Comment:
This you can keep the same without making the switch statement any more complicated.

Reply:
Its even more complicated to understand when I keep it. Because then I'll have to add an if-statements later to distinguish between compressible/incompressible when actually allocating. This way you see immediately see what will be allocated (i.e. what boolean is set to true).

But anyway. All that this shows is that we have to do something about it at some point and neither approach adds or removes much readability. @rsanfer and myself have thought already about moving some the allocations to the iteration structure, as the only that class essentially has to know what is in the solver container. Likewise, the solver class is the only class that has to know what is in the numerics class and so on.

Reply:
And ideally own the actual container they instantiate.